lib/truffle/io/console.rb

```diff
@@ -45,6 +45,38 @@ def getpass(prompt = nil)
 
 
 class IO
+  def self.console(sym=nil, *args)
+    raise TypeError, "expected Symbol, got #{sym.class}" unless sym.nil? || sym.kind_of?(Symbol)
+
+    # klass = self == IO ? File : self
+    if defined?(@console) # using ivar instead of hidden const as in MRI
+      con = @console
+      # MRI checks IO internals : (!RB_TYPE_P(con, T_FILE) || (!(fptr = RFILE(con)->fptr) || GetReadFD(fptr) == -1))
+      if !con.kind_of?(File) || (con.kind_of?(IO) && (con.closed? || !FileTest.readable?(con)))
+        remove_instance_variable :@console
+        con = nil
+      end
+    end
+
+    if sym
+      if sym == :close
+        if con
+          con.close
+          remove_instance_variable :@console if defined?(@console)
+        end
+        return nil
+      end
+    end
+
+    if !con && $stdin.tty?
+      con = File.open('/dev/tty', 'r+')
+      con.sync = true
+      @console = con
```

Comment:
This is obviously not thread-safe, not sure if you care in this case.

Reply:
I'm not too concerned about performance of this method. It's probably best to make it synchronized.

Reply:
That would work, although the `send` below shouldn't be part of the synchronize block.